[Bug]: Hop Server no longer can be started by name

### Apache Hop version?

2.4.0

### Java version?

openjdk version "11.0.17" 2022-10-18

### Operating system

Linux

### What happened?

When starting a hop server by name and by referencing an environment, I get a nullpointer exception:

`matt@canis:/opt/hop$ sh hop-server.sh -e 'IT hop_server' -n testserver
2023/02/16 13:47:50 - HopServer - Referencing environment 'IT hop_server' for project IT hop_server' in 
2023/02/16 13:47:50 - HopServer - Enabling project 'IT hop_server'
Error found during execution!
picocli.CommandLine$ExecutionException: There was an error during the startup of the Hop server
        at org.apache.hop.www.HopServer.run(HopServer.java:312)
        at org.apache.hop.www.HopServer.main(HopServer.java:575)
Caused by: java.lang.NullPointerException
        at java.base/java.util.Objects.requireNonNull(Objects.java:221)
        at java.base/java.util.ImmutableCollections$ListN.<init>(ImmutableCollections.java:432)
        at java.base/java.util.List.of(List.java:842)
        at org.apache.hop.www.HopServer.setupByServerName(HopServer.java:350)
        at org.apache.hop.www.HopServer.run(HopServer.java:256)
        ... 1 more
`

### Issue Priority

Priority: 1

### Issue Component

Component: Hop Server

--- a/docs/hop-user-manual/modules/ROOT/pages/hop-server/index.adoc
+++ b/docs/hop-user-manual/modules/ROOT/pages/hop-server/index.adoc
@@ -42,37 +42,40 @@ On Windows, this is `hop-server.bat`, on Mac and Linux, run `hop-server.sh`.
 [source,bash]
 ----
 Usage: <main class> [-k] [-gs] [-e=<environmentOption>] [-id=<id>]
-                    [-j=<projectOption>] [-l=<level>] [-p=<password>]
-                    [-ps=<pipelineName>] [-u=<username>] [-ws=<workflowName>]
-                    [-s=<systemProperties>[,<systemProperties>...]]...
-                    <parameters>...
-      <parameters>...   One XML configuration file or a hostname and port
+                    [-j=<projectOption>] [-l=<level>] [-n=<serverName>]
+                    [-p=<password>] [-ps=<pipelineName>] [-u=<username>]
+                    [-ws=<workflowName>] [-s=<systemProperties>[,
+                    <systemProperties>...]]... [<parameters>...]
+      [<parameters>...]   One XML configuration file or a hostname and port
   -e, --environment=<environmentOption>
-                        The name of the lifecycle environment to use
-      -gs, --general-status
-                        List the general status of the server
-      -id=<id>          Specify the ID of the pipeline or workflow to query
+                          The name of the lifecycle environment to use
+  -gs, --general-status
+                          List the general status of the server
+      -id=<id>            Specify the ID of the pipeline or workflow to query
   -j, --project=<projectOption>
-                        The name of the project to use
-  -k, --kill            Stop the running hopServer server.  This is only allowed
-                          when using the hostname/port form of the command. Use the
-                          -s and -u options to authenticate
-  -l, --level=<level>   The debug level, one of NOTHING, ERROR, MINIMAL, BASIC, DETAILED, DEBUG, ROWLEVEL
+                          The name of the project to use
+  -k, --kill              Stop the running hopServer server.  This is only
+                            allowed when using the hostname/port form of the
+                            command. Use the -s and -u options to authenticate
+  -l, --level=<level>     The debug level, one of NOTHING, ERROR, MINIMAL,
+                            BASIC, DETAILED, DEBUG, ROWLEVEL
+  -n, --server-name=<serverName>
+                          The name of the server to start as defined in the
+                            metadata.
   -p, --password=<password>
-                        The server password.  Required for administrative operations
-                          only, not for starting the server.
-      -ps, --pipeline-status=<pipelineName>
-                        List the status of the pipeline with this name (also specify
-                          the -id option)
+                          The server password.  Required for administrative
+                            operations only, not for starting the server.
+  -ps, --pipeline-status=<pipelineName>
+                          List the status of the pipeline with this name (also
+                            specify the -id option)
   -s, --system-properties=<systemProperties>[,<systemProperties>...]
-                        A comma separated list of KEY=VALUE pairs
+                          A comma separated list of KEY=VALUE pairs
   -u, --userName=<username>
-                        The server user name.  Required for administrative
-                          operations only, not for starting the server.
-      -ws, --workflow-status=<workflowName>
-                        List the status of the workflow with this name (also specify
-                          the -id option)
-
+                          The server user name.  Required for administrative
+                            operations only, not for starting the server.
+  -ws, --workflow-status=<workflowName>
+                          List the status of the workflow with this name (also
+                            specify the -id option)
 ----
 
 The available Hop Server options are:
@@ -109,6 +112,10 @@ Specify a comma separated list of KEY=VALUE pairs.
 |--environment
 |The name of the project lifecycle environment to enable before startup.
 This is provided for by the 'projects' plugin.
+
+|-n
+|--server-name
+|Name of the server metadata object to start, has to be used in combination with -e to know which environment to load it from
 
 |-j
 |--project

--- a/engine/src/main/java/org/apache/hop/www/HopServer.java
+++ b/engine/src/main/java/org/apache/hop/www/HopServer.java
@@ -336,7 +336,10 @@ public class HopServer implements Runnable, IHasHopMetadataProvider {
     String hostname = variables.resolve(hopServer.getHostname());
     String port = variables.resolve(hopServer.getPort());
     String shudownPort = variables.resolve(hopServer.getShutdownPort());
-    parameters = List.of(hostname, port, shudownPort);
+    parameters = List.of(hostname, port);
+    if (StringUtils.isNotEmpty(shudownPort)) {
+      parameters.add(shudownPort);
+    }
   }
 
   private boolean handleQueryOptions() {

